sankey.py

```diff
@@ -69,19 +69,32 @@ def sankey(left, right, leftWeight=None, rightWeight=None, colorDict=None,
     plt.rc('font', family='serif')
 
     # Create Dataframe
+    if isinstance(left, pd.Series):
+        left = left.reset_index(drop=True)
+    if isinstance(right, pd.Series):
+        right = right.reset_index(drop=True)
     df = pd.DataFrame({'left': left, 'right': right, 'leftWeight': leftWeight,
                        'rightWeight': rightWeight}, index=range(len(left)))
+    
+    if len(df[(df.left.isnull()) | (df.right.isnull())]):
+        raise RuntimeError('Sankey graph does not support null values.')
 
     # Identify all labels that appear 'left' or 'right'
     allLabels = pd.Series(np.r_[df.left.unique(), df.right.unique()]).unique()
 
     # Identify left labels
     if len(leftLabels) == 0:
         leftLabels = pd.Series(df.left.unique()).unique()
+    else:
+        if set(leftLabels) != set(df['left'].unique().tolist()):
+            raise RuntimeError('Left labels and left data do not match.')
 
     # Identify right labels
     if len(rightLabels) == 0:
         rightLabels = pd.Series(df.right.unique()).unique()
+    else:
+        if set(rightLabels) != set(df['right'].unique().tolist()):
+            raise RuntimeError('Right labels and right data do not match.')
```

Comment:
Same remark than above, but this how about putting both the check in a generic function like "check_matching_labels_and_data()" ?

Reply:
Abstracting to a separate function is probably a good idea.